dulwich/server.py

```diff
@@ -1018,6 +1018,14 @@ def handle(self):
         # backend can now deal with this refs and read a pack using self.read
         status = self._apply_pack(client_refs)
 
+        hook = self.repo.hooks.get('post-receive', None)
+        if hook:
+            hook.execute(
+                stdin='\n'.join([' '.join(i) for i in client_refs])
```

Comment:
Please just pass in the refs; the hook can deal with formatting them for the shell script.

This makes it easier for non-shell hooks to be implemented.